### PR TITLE
Augury tests

### PR DESCRIPTION
**Pull request by svevang**
closes https://github.com/PRX/internal/issues/1082
Adds two augury integration tests

1) Runs a report on the availability endpoint
2) Travel to / runs a campaign report

### CODE REVIEW

File: cypress/e2e/augury.cy.ts [hunk 1]
```diff
@@ -81,4 +83,70 @@ describe("Augury", () => {
...
+    getStartDateInput(): Chainable<JQuery<HTMLElement>>;
+    getEndDateInput(): Chainable<JQuery<HTMLElement>>;
+    getInventoryTypeSelect(): Chainable<JQuery<HTMLElement>>;
+    getAvailabilityResults(): Chainable<JQuery<HTMLElement>>;
+    selectSlimSelect(selector: string, value: string): Chainable<JQuery<HTMLElement>>;
+  }
+}
+
+// Custom commands that can be chained
+Cypress.Commands.add("getStartDateInput", () => {
```
Comment (cavis) [inline cypress/e2e/augury.cy.ts:135]: Neat, I've not used these!

I'm curious - do these become global commands?  Like, would this be available in the `feeder.cy.ts` file?

File: cypress/e2e/augury.cy.ts [hunk 2]
```diff
@@ -81,4 +83,70 @@ describe("Augury", () => {
...
+    cy.getAvailabilityResults().should("be.visible");
+    cy.getAvailabilityResults().contains("Calculate");
+    cy.getAvailabilityResults().click();
+
+    // page should container "Forecasted Inventory"
+    cy.contains("Forecasted Inventory");
+  });
+
+  it("generates a campaign report", () => {
+    // Assuming we have a campaign with ID 1
```
Comment (cavis) [inline cypress/e2e/augury.cy.ts:105]: Hrm... where does it assume this ID?

File: .eslintrc [hunk 3]
```diff
@@ -8,6 +8,10 @@
     "ecmaVersion": "latest",
     "sourceType": "module"
   },
+  "rules": {
+    "no-use-before-define": "off",
+    "no-unused-vars": "off"
```
Comment (cavis) [inline .eslintrc:13]: I'm not sure I follow which things in your test needed this.  The `declare namespace Cypress` thing perhaps?

**Review by cavis — APPROVED** [review]
👍 a couple questions, but this all looks good to me.

File: .eslintrc (hunk 3 above)
Comment (svevang) [reply .eslintrc:13]: With these turn off, the eslint errors I get are:

```
    5:14  error  'Cypress' was used before it was defined  no-use-before-define
    5:44  error  'Cypress' was used before it was defined  no-use-before-define
    6:5   error  'Cypress' was used before it was defined  no-use-before-define
    6:42  error  'Cypress' was used before it was defined  no-use-before-define
  124:23  error  'Subject' is defined but never used       no-unused-vars
  129:22  error  'selector' is defined but never used      no-unused-vars
  129:40  error  'value' is defined but never used         no-unused-vars
```

I believe this is related to the mix of tsc + commonjs + cypress that we're using.  I don't think these linting errors are correct.

Thoughts?

File: cypress/e2e/augury.cy.ts (hunk 1 above)
Comment (svevang) [reply cypress/e2e/augury.cy.ts:134]: I would assume global, but did not get too far into it.  I'd expect some global accessor methods like these could come in handy with the particular markup that's shared between the apps.

File: .eslintrc (hunk 3 above)
Comment (svevang) [reply .eslintrc:13]: Adding `/* global Cypress */` to the top results in:

`error  'Cypress' is already defined as a built-in global variable`

So somehow it's already defined as a global and is used before it's defined.

Comment (cavis) [reply .eslintrc:13]: Weird.  Yeah, I'm fine with the change.

**Pull request merged**